### DISCUSSION TITLE
How to conditionally add styles for specific files

### DISCUSSION THREAD

**Discussion by tylersayshi** [tailwindlabs/tailwindcss#5175] [Help]
I am currently creating a component library where I am using tailwind & storybook. Currently, I am using the JIT mode and any tailwind classes that I add to *.stories.tsx files get put into the final styles for my storybook. This is good for when I am running storybook or building it, but when I run build for the component library I would like to exclude any classes in .stories.tsx files.

My first attempt to do this was to try to exclude stories files in the purge field like this:

```js
purge: ['./src/**/!(*.stories).{ts,tsx}'],
``` 

If this worked I would create an environment variable for when storybook is being used to add those files as well

```js
purge: [`./src/**/${process.env.SB ? '*' : '!(*.stories)'}.{ts,tsx}`],
```

Unfortunately, the glob that was meant to exclude stories files was not doing what I was hoping it would and was still including classes in my final build from .stories.tsx files. 

Does anyone know if I am missing something? Is there something else I need to set to exclude certain files? Thanks in advance for any help.

Edit: The specific issue here is that `'./src/**/!(*.stories).{ts,tsx}'` is still including files with the `.stories.tsx` extension.

**Comment by simonswiss** [comment]
Hey!

Have you tried rewriting your ternary logic "outside" of the glob?

```js
purge: process.env.SB ? ['./src/**/*.{ts,tsx}'] : ['./src/**/!(*.stories).{ts,tsx}']
```

You can also probably use the `NODE_ENV` environment variable that many build tools set to `production` for production build, instead of creating an extra environment variable.

**Reply by tylersayshi** [reply]
That does look cleaner, but I don't think I was exactly clear about the problem I am having. Anytime I use this glob `'./src/**/!(*.stories).{ts,tsx}'` the tailwind classes in files with `.stories` are still included in the tailwind css output. It seems that the `!(*.stories)` is being completely ignored for some reason.
